File: 28.py
def subsetsWithTarget(nums, target):
    nums.sort()
    res = []

    def backtrack(start, path, curr_sum):
        if curr_sum == target:
            res.append(path[:])
            return
        if curr_sum > target:
            return

        for i in range(start, len(nums)):
            if i > start and nums[i] == nums[i-1]:
                continue
            backtrack(i + 1, path + [nums[i]], curr_sum + nums[i])

    backtrack(0, [], 0)
    return res

File: 28.js
function subsetsWithTarget(nums, target) {
    nums.sort((a, b) => a - b);
    const res = [];

    function backtrack(start, path, sum) {
        if (sum === target) {
            res.push([...path]);
            return;
        }
        if (sum > target) return;

        for (let i = start; i < nums.length; i++) {
            if (i > start && nums[i] === nums[i - 1]) continue;
            path.push(nums[i]);
            backtrack(i + 1, path, sum + nums[i]);
            path.pop();
        }
    }

    backtrack(0, [], 0);
    return res;
}
